### PR TITLE
Remove automatic C2D link open and remove timeout on internal retry

### PR DESCRIPTION
**Pull request by CIPop**
- Removing automatic C2D creation
-  modifying subscription internal retry (will retry until object is closed/disposed)
-  fixing some of the object lifetime and locking

### CODE REVIEW

File: iothub/device/src/Transport/Amqp/AmqpConnectionHolder.cs [hunk 1]
```diff
@@ -183,7 +194,7 @@ private void Shutdown()
         {
             if (Logging.IsEnabled) Logging.Enter(this, AmqpConnection, $"{nameof(Shutdown)}");
             AmqpAuthenticationRefresher?.StopLoop();
-            AmqpLinkHelper.CloseAmqpObject(AmqpConnection);
+            AmqpConnection.Abort();
```
Comment (davilu) [inline iothub/device/src/Transport/Amqp/AmqpConnectionHolder.cs]: missing `?`?

File: iothub/device/src/InternalClient.cs [hunk 2]
```diff
@@ -125,6 +125,7 @@ public enum MessageResponse { None, Completed, Abandoned };
     internal class InternalClient : IDisposable
     {
         private uint _operationTimeoutInMilliseconds = DeviceClient.DefaultOperationTimeoutInMilliseconds;
+        private uint _retryTimeoutInMilliseconds = 0;
```
Comment (zolvarga) [inline iothub/device/src/InternalClient.cs]: >_retryTimeoutInMilliseconds [](start = 21, length = 27)

I don't see where this one used.

**Comment by CIPop** [comment]
@davilu @abhipsaMisra /cc @prmathur-microsoft - got an AMQP token refresh failure with my run that passed last Friday.

**Pull request merged**